### PR TITLE
IGNITE-12886 Introduce separate SQL configuration

### PR DESCRIPTION
**Pull request by tledkov-gridgain**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java [hunk 1]
```diff
@@ -313,12 +320,6 @@
     /** Index create/rebuild pool size. */
     private int buildIdxPoolSize = DFLT_BUILD_IDX_THREAD_POOL_SIZE;
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java:321]: Should this be a part of SqlConfiguration as well?

File: modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    /** SQL query history size. */
+    private int sqlQryHistSize = DFLT_SQL_QUERY_HISTORY_SIZE;
+
+    /**
+     * Defines the default query timeout.
+     *
+     * Defaults to {@link #DFLT_QRY_TIMEOUT}.
+     * {@code 0} means there is no timeout (this
+     * is a default value)
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java:53]: Let's remove linefeed here to make more readable.

File: modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+     * @return Default query timeout.
+     */
+    public long getDefaultQueryTimeout() {
+        return dfltQryTimeout;
+    }
+
+    /**
+     * Sets timeout in milliseconds for default query timeout.
+     * {@code 0} means there is no timeout (this
+     * is a default value)
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java:64]: Let's remove linefeed here to make more readable.

File: modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java (hunk 1 above)
Comment (tledkov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java:321]: Not agreed. All we configure and start all executor services at the IgnitionEx.IgniteNamedInstance#start0. So it looks looks like a core feature.

File: modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java (hunk 2 above)
Comment (tledkov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java (hunk 3 above)
Comment (tledkov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/configuration/SqlConfiguration.java]: Fixed

**Pull request closed without merging**